### PR TITLE
Actually enable flyspell-prog-mode

### PR DESCRIPTION
**Pull request by lunaryorn**
Add `flycheck-prog-mode` to `prog-mode-hook` to actually turn it on in programming modes.

### CODE REVIEW

File: layers/spell-checking/packages.el [hunk 1]
```diff
@@ -31,14 +34,12 @@
...
         "Enable automatic spell checking."
         :evil-leader "tS")
+
       (evil-leader/set-key
         "Sd" 'ispell-change-dictionary
         "Sn" 'flyspell-goto-next-error))
     :config
-    (progn
-      (when spell-checking-enable-by-default
-        (flyspell-prog-mode))
```
Comment (lunaryorn) [inline layers/spell-checking/packages.el]: I'm not sure what _this_ was supposed to do… `flyspell-prog-mode` is a wrapper around `flyspell-mode`, i.e. effectively a buffer local minor mode—calling it this way has no effect at all.

**Comment by robbyoconnor** [comment]
:+1: nice!

**Comment by zilongshanren** [comment]
Once it is merged, I could remove it from my config now. :+1:

**Comment by d12frosted** [comment]
Oh nice. The same thing.

**Comment by syl20bnr** [comment]
Excellent, I was looking for this myself, thank you :+1: 
Cherry-picked into `develop` branch, you can safely delete your branch.

**Pull request closed without merging**